src/System.Private.CoreLib/src/System/Threading/Timer.Unix.cs

```diff
@@ -13,109 +14,127 @@ namespace System.Threading
     //
     // Unix-specific implementation of Timer
     //
-    internal partial class TimerQueue
+    internal partial class TimerQueue : IThreadPoolWorkItem
     {
+        private static List<ScheduledTimer> s_scheduledTimers;
+
         /// <summary>
         /// This event is used by the timer thread to wait for timer expiration. It is also
         /// used to notify the timer thread that a new timer has been set.
         /// </summary>
-        private AutoResetEvent _timerEvent;
+        private static readonly AutoResetEvent s_timerEvent = new AutoResetEvent(false);
 
-        /// <summary>
-        /// This field stores the value of next timer that the timer thread should install.
-        /// </summary>
-        private volatile int _nextTimerDuration;
+        private readonly int _id;
 
         private TimerQueue(int id)
         {
+            _id = id;
+        }
+
+        private static List<ScheduledTimer> InitializeScheduledTimerManager_Locked()
+        {
+            Debug.Assert(s_lock.IsAcquired);
+            Debug.Assert(s_scheduledTimers == null);
+
+            var scheduledTimers = new List<ScheduledTimer>(Instances.Length);
+
+            RuntimeThread timerThread = RuntimeThread.Create(TimerThread);
+            timerThread.IsBackground = true;
+            timerThread.Start();
+
+            // Do this after creating the thread in case thread creation fails so that it will try again next time
+            s_scheduledTimers = scheduledTimers;
```

Comment:
If you move this above the thread creation then the lock on line 78 becomes unnecessary.

Reply:
Ah, nvm, I type faster than read... :)